File: ruby/sorting/bubblesort.rb
def bubble(arry)
  swap = true
  l = arry.length
  while swap do
    swap = false
    (l-1).times do |n|
      if arry[n] > arry[n+1]
        arry[n], arry[n + 1] = arry[n + 1], arry[n]
        swap = true
      end
    end
  end
  arry
end

arry = [5,4,6,2,1,73,4,3,5,6,2,3,6,20,3,2,1]
p bubble(arry)

File: js/sorting/bubblesort.js
function bubble(arry) {
  let swap = false
  do {
    swap = false;
    for(let i = 0; i < arry.length; i++) {
      if(arry[i] > arry[i + 1]) {
        console.log(arry)
        let change = arry[i];
        arry[i] = arry[i+1];
        arry[i+1] = change;
        swap = true;
      }
    }
  } while(swap);
  return arry
}

console.log(bubble([20,4,3,5,3,2,5,1]))
